server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -343,4 +370,46 @@ public void collect(GarbageCollectionEnvironment gce)
       deleteConfirmed(gce, candidateMap);
     }
   }
+
+  /**
+   * Track metadata rows read to help validate that gc scan has complete information to make a
+   * decision on deleting files
+   */
+  private static class MetadataReadTracker {
```

Comment:
I like this class.  I was wondering if more of the logic behind the check could be moved into this class and out of the loop.  I experimented locally and that resulted in [this change](https://github.com/keith-turner/accumulo/commit/367dadc837c9886eac1d5030d68e198edbbd45e0).  It's a stylistic change, so feel free to ignore.

Reply:
I merged your changes into the branch (I hope).  Thanks!

Reply:
ok, I didn't format the code or run the test you added.  I just wanted to quickly see if it was workable.